### PR TITLE
Capture exceptions

### PR DESCRIPTION
**Pull request by dan437**
## Explanation
We capture now exceptions in 2 places in the `timedoutGasReturn` function.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: app/scripts/controllers/swaps.js [hunk 1]
```diff
@@ -492,13 +495,21 @@ export default class SwapsController {
...
 
-  timedoutGasReturn(tradeTxParams) {
+  timedoutGasReturn(tradeTxParams, aggregator = '') {
     return new Promise((resolve) => {
       let gasTimedOut = false;
 
       const gasTimeout = setTimeout(() => {
         gasTimedOut = true;
-        resolve({ gasLimit: null, simulationFails: true });
+        captureException(new Error('timedoutGasReturn: timeout'), {
```
Comment (Gudahtt) [inline app/scripts/controllers/swaps.js:504]: It seems like this error represents a timeout, which is something that can happen in the normal operation of the wallet (e.g. in poor network conditions). We use Sentry just for capturing errors that should never happen. We're making great efforts to reduce our overall Sentry volume with the intention to eventually bring it down to roughly zero, but we'll never get there if we add these sort of errors.

If something is expected to fail, but you want more information about when and how it fails, I'd suggest using a metric event instead.  Then we can track the behavior in Mixpanel.

Comment (dan437) [reply app/scripts/controllers/swaps.js:504]: That sounds great, will do that for this error.

File: shared/constants/metametrics.ts [hunk 2]
```diff
@@ -550,6 +550,7 @@ export enum MetaMetricsEventName {
   PortfolioLinkClicked = 'Portfolio Link Clicked',
   ProviderMethodCalled = 'Provider Method Called',
   PublicAddressCopied = 'Public Address Copied',
+  QuoteError = 'Quote Error',
```
Comment (brad-decker) [inline shared/constants/metametrics.ts:553]: Out of curiosity, were these new event properties confirmed through the metrics team / @worldlyjohn

Comment (dan437) [reply shared/constants/metametrics.ts:553]: Yup, we checked it with @worldlyjohn

**Review by brad-decker — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**